### PR TITLE
Cleaned up and slightly expanded the readme

### PR DESCRIPTION
**Pull request by mars888**
Tried to clean up some of the language and structure.

### CODE REVIEW

**Review by Corniel — APPROVED** [review]
LGTM!

File: README.md [hunk 1]
```diff
@@ -8,42 +8,58 @@
...
-It should be provided with an [IValidator<TAggragate>](https://github.com/Qowaiv/qowaiv-validation)
-implementation of choice.
+An Aggregate Root is the root of every Domain within a DDD context. When
+implementing a new aggregate root there are several steps that have to be taken.
 
-Furthermore there is an option to have a base where identity and the handling of
-the events that should be published (as they do not lead to an invalid state):
-`AggregateRoot<TAggregate>`. It has an `protected abstract void AddEventsToBuffer(params object[] events)`
-method that has to be overridden, to achieve the persistence of the events.
+First a new Aggregate Root should have a corresponding
```
Comment (LauraXiulan) [inline README.md:34]: Add comma after 'First'

File: README.md [hunk 2]
```diff
@@ -8,42 +8,58 @@
...
+- `AggregateRoot<TAggregate>`
+- `AggregateRoot<TAggregate, TId>`
+
+`AggregateRoot<TAggregate>` is a low level base class that provides a framework
+for handling the application of events. It has no event store of its own. Implementations
+of this class should override the `protected abstract void AddEventsToBuffer(params object[] events)`
+method to achieve persistence of events.
+
+The second option, `AggregateRoot<TAggregate, TId>`, inherits from
+`AggregateRoot<TAggregate>`. It has build-in identity support and systems for
```
Comment (LauraXiulan) [inline README.md:51]: buil**t**-in

File: README.md [hunk 3]
```diff
@@ -8,42 +8,58 @@
...
+storing events in the integrated `EventBuffer<TId>`. Events that
+should be persisted are added to it this buffer, and it can return both the committed as
```
Comment (LauraXiulan) [inline README.md:53]: remove 'it' or 'this buffer'

File: README.md [hunk 4]
```diff
@@ -112,19 +128,20 @@ public class FinancialEntryValidator : FluentModelValidator<FinancialEntry>
...
 
-You could argue that some of the rules specified in the validator should be handled
-as part of the *anti-corruption layer*, but that is another topic. The point is
-that by defining those constraints, you can no longer add any entry line with
-an other currency then the lines already added, nor have an financial entry that
-is not balanced, both extremely important in this domain.
+You could argue that some of the rules specified in the validator should be
+handled as part of the *anti-corruption layer*, but that is another topic. The
+point is that by defining those constraints, you can no longer add any entry
+line with any other currency then the lines already added, nor add any financial
```
Comment (LauraXiulan) [inline README.md:134]: 'than' instead of 'then'

**Review by LauraXiulan — APPROVED** [review]

**Comment by Corniel** [comment]
@mars888 Can you apply @LauraXiulan 's suggestions?

**Comment by mars888** [comment]
Updated, thanks @LauraXiulan!

**Pull request merged**